GitUI/CommandsDialogs/BrowseDialog/FormCommitCount.cs

```diff
@@ -35,28 +37,23 @@ private void FetchData()
             _backgroundLoaderTokenSource = new CancellationTokenSource();
             var token = _backgroundLoaderTokenSource.Token;
 
-            ThreadHelper.JoinableTaskFactory.RunAsync(
-                async () =>
-                {
-                    Loading.Visible = true;
+            Loading.Visible = true;
 
-                    CommitCount.Text = "";
+            CommitCount.Text = "";
 
-                    var includeSubmodules = cbIncludeSubmodules.Checked;
+            var includeSubmodules = cbIncludeSubmodules.Checked;
 
-                    await TaskScheduler.Default;
+            await TaskScheduler.Default;
 
-                    var text = GenerateText(Module, includeSubmodules, token);
+            var text = GenerateText(Module, includeSubmodules, token);
 
-                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(token);
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(token);
 
-                    if (!token.IsCancellationRequested)
-                    {
-                        CommitCount.Text = text;
-                        Loading.Visible = false;
-                    }
-                })
-                .FileAndForget();
+            if (!token.IsCancellationRequested)
```

Comment:
❕ Needs a check for `IsDisposed`

:memo: I'll likely update our custom `SwitchToMainThreadAsync` extension method to support a cancellation token, allowing it to be used in places like this.

Reply:
Thanks, done.